From e623b50531fcab853e7591ccb5d3f3fe01907b3c Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 23 Jan 2023 10:59:13 +0200
Subject: [PATCH] [SW-117389] ppboot: add comment for pcie elbi mem access

This patch adds the reason why pcie elbi access is done
after ppboot_tpm_extend - no functional change.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I053129319d189ec608a908068af6cbe271bf6f66
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index 630834b..bbffefa 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -337,6 +337,11 @@
 	ppboot_security_init();
 	hl_pcie_init();
 	ppboot_tpm_extend();
+	/*
+	 * Enabling pcie elbi after ppboot_tpm_extend since elbi protection level
+	 * is usually secured and it can access configuration space. This access
+	 * can interfere with tpm extend.
+	 */
 	if (link_is_up)
 		hl_pcie_enable_elbi_access();
 	ppboot_time_critical_section_end();
